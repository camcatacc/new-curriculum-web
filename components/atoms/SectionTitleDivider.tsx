import { Divider } from "@material-ui/core";
import React from "react";
import { black } from "../../styles/customColors";
import { InlineStyles } from "../../styles/stylesInterface";

export const SectionTitleDivider = () => {
    return (
        <div
            style={{ borderColor: black }}
            className="self-center border-b-4 border-solid w-20 mt-4 mb-4"
        />
    );
};
