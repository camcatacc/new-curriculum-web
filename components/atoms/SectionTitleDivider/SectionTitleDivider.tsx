// Modules
import { motion } from "framer-motion";
import React from "react";
import { useTheme } from "@material-ui/core";

// Utils
import { getMotionFromOutsideVariants } from "utils/getMotionFromOutsideOptions";

// Definitions
import type { MotionProps } from "framer-motion";
export interface SectionTitleDividerProps {
	animated?: boolean;
	delay?: number;
}

// Auxiliary functions
const getMotionOptions = (animated?: boolean, delay?: number): MotionProps => {
	if (!animated) return {};
	const { initial, animation } = getMotionFromOutsideVariants("fromLeft", delay);
	return {
		variants: { initial, animation },
		initial: "initial",
		whileInView: "animation",
		viewport: {
			once: true,
			amount: 1
		}
	};
};

// Element
const SectionTitleDivider = ({ animated, delay }: SectionTitleDividerProps) => {
	const theme = useTheme();
	const motionOptions = getMotionOptions(animated, delay);
	return <motion.div {...motionOptions} className={`self-center border-b-2 ${theme.palette.type === "dark" ? "border-white" : "border-black"} border-solid w-20 mt-4 mb-4`} />;
};

export default SectionTitleDivider;
