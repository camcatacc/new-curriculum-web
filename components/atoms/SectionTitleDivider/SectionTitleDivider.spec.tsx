// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import SectionTitleDivider from "components/atoms/SectionTitleDivider/SectionTitleDivider";

// Tests
describe("SectionTitleDivider", () => {
	it("Renders as expected", () => {
		const { container } = render(<SectionTitleDivider />);
		expect(container).toMatchSnapshot();
	});
});
