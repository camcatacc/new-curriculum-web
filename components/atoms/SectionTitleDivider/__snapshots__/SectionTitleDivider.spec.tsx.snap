// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitleDivider Renders as expected 1`] = `
<div>
  <div
    class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
  />
</div>
`;
